From 90a82d55f6a393f8f93562a9062f52f163903327 Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Wed, 2 Nov 2011 16:06:49 +0800
Subject: [PATCH 062/641] UniCore64: Fix return type of pte_alloc_one function

Fixed warnings:
mm/memory.c: In function '__pte_alloc':
mm/memory.c:598:18: warning: initialization from incompatible pointer type
mm/vmalloc.c: In function 'vunmap_pte_range':
mm/vmalloc.c:40:6: warning: assignment makes pointer from integer without a cast
mm/memory.c: In function 'copy_pte_range':
mm/memory.c:941:12: warning: initialization makes pointer from integer without a cast
mm/memory.c:944:10: warning: assignment makes pointer from integer without a cast
mm/memory.c: In function 'zap_pte_range':
mm/memory.c:1121:14: warning: initialization makes pointer from integer without a cast
mm/mremap.c: In function 'move_ptes':
mm/mremap.c:103:12: warning: initialization makes pointer from integer without a cast
mm/mremap.c:104:10: warning: assignment makes pointer from integer without a cast

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/pgalloc.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/include/asm/pgalloc.h b/arch/unicore64/include/asm/pgalloc.h
index 2a836e3..7ffe0cc 100644
--- a/arch/unicore64/include/asm/pgalloc.h
+++ b/arch/unicore64/include/asm/pgalloc.h
@@ -4,7 +4,7 @@
 extern pgd_t *pgd_alloc(struct mm_struct *);
 extern void pgd_free(struct mm_struct *, pgd_t *);
 
-extern pgtable_t *pte_alloc_one(struct mm_struct *, unsigned long);
+extern pgtable_t pte_alloc_one(struct mm_struct *, unsigned long);
 extern pte_t *pte_alloc_one_kernel(struct mm_struct *, unsigned long);
 
 extern void pte_free(struct mm_struct *, pgtable_t);
-- 
1.7.9.5

